src/app/content/highlights/components/SummaryPopup/utils.ts

```diff
@@ -0,0 +1,11 @@
+import { Highlight } from '@openstax/highlighter/dist/api';
+import { generateTargetQueryFromScrollTarget } from '../../../../navigation/utils';
+import { Book } from '../../../types';
+import { findArchiveTreeNode } from '../../../utils/archiveTreeUtils';
+import { getBookPageUrlAndParams } from '../../../utils/urlUtils';
+
+export const createHighlightLink = ({ sourceId, id, anchor: elementId }: Highlight, book?: Book) => {
+  const page = book ? findArchiveTreeNode(book.tree, sourceId) : undefined;
+  const target = generateTargetQueryFromScrollTarget({ id, type: 'highlight', elementId });
+  return `${page && book ? getBookPageUrlAndParams(book, page).url : ''}?${target}`;
```

Comment:
instead of generating the actual url here, you could generate the props to set on the ContentLink component, that might be a little simpler

Reply:
I feel like `<ContentLink>` is already too heavy and it is hard to manage

Reply:
i don't understand, i don't think you'd have to change contentlink to use it for this, it already does all but the fist line of this function

you could rename this function `createHighlightLinkProps`, have it return the book, page, and scrolltarget, and just apply those to contentlink

Reply:
`createHighlightLink` requires `Book` to be passed to it - I think there is no way to get `Book` for `Highlight` synchronously, is there?
Anyway, since `<ContentLink>` requires `Book` and `Page` to be passed in and already accepts `scrollTarget` I think it has everything to create a link to the highlight  

However, this function (`createHighlightLink`) is used to pass a link to the `<ContextMenu>` and renders a link like this:
```jsx
<DropdownItem
  data-testid='go-to-highlight'
  message='i18n:highlighting:dropdown:go-to-highlight'
  prefix={<StyledLinkIcon/>}
  href={linkToHighlight}
  target='_blank'
/>
```
Would you like me to change this `<DropdownItem>` to custom styled `<ContentLink>`?